### PR TITLE
Fixes typos, adds header specifications for etag and a few other options. No new API added here

### PR DESCRIPTION
**Pull request by mastermanu**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by andrew-buckley** [comment]
This is related to #1723. These changes were discovered after the merge.

File: specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json [hunk 1]
```diff
@@ -887,7 +894,7 @@
         "tags": [ "Certificates" ],
         "summary": "Get the certificate list.",
         "description": "Returns the list of certificates.",
-        "operationId": "Certificates_ListByIoTHub",
+        "operationId": "Certificates_ListByIotHub",
```
Comment (dsgouda) [inline specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: Just curious, what issues did you notice with AutoRest due to this.

**Review by dsgouda — COMMENTED** [review]
Looks good for the most part, just curious about the AutoRest issues.

File: specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json (hunk 1 above)
Comment (andrew-buckley) [reply specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: It's kind of funny. If we leave it how it was previously spelled, at least the python generator was giving me the function "list_by_io_thub", changing the casing properly gives me "list_by_iot_hub".

Comment (dsgouda) [reply specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: I see, I think we are extracting words that start with a capitalized letter or grouping together capitalized letters into a single word, for most cases this is a viable solution while `IoT` looks like an exception. I guess this is a preferred workaround.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**